openpype/hosts/blender/hooks/pre_add_set_scene_properties.py

```diff
@@ -0,0 +1,105 @@
+import os
+import tempfile
+
+from openpype.hosts.blender.hooks import pre_add_run_python_script_arg
+from openpype.lib import PreLaunchHook
+
+
+class SetSceneProperties(PreLaunchHook):
```

Comment:
Is this hook always trigger when opening a Blender scene? Because since you set `target_os = "linux"` does this means all paths are reformatted even when working on Windows ?

Reply:
This property only refers to the os on which we want to render our scene. So it doesn't depend from the working file os, but rather the one on the render computers. It should be updated / better handled if theses machines changes to another os, but it seems unlikely.
If you prefer, we could update this variable name to reflect this use, like `render_target_os` maybe ?

Reply:
To add some precisions : the formatted ouputpath is calculated at each scene opening, but the set only happens when the temporary files are generated, when the addon is triggered (so it should never have any impact on the working files)

Reply:
So this `PreLaunchHook` is triggered when Blender start, because a `PreLaunchHook` is always processed before application is launched.
Then inside the execute method you compute the `render-layer-path` and `output-path` and then you add these values as default script args?

What does this code do? Adding default args for ALL the blender scripts?
```python
self.launch_context.data.setdefault("script_args", []).extend(
            [
                '--render-layer-path',
                render_layer_path,
                '--output-path',
                output_path

            ]
        )
```

Reply:
Yes, it triggers when we launch Blender from Openpype's launcher and is executed before Blender is launched.
The role of this hook is to compute or transmit some necessary values given by OpenPype (typically data about projects or shots corresponding to the working file). These values are injected as arguments when launching the Blender executable. Then, the given startup scripts pick the arguments they need. The concerned script here is the `set_scene_properties` whose role is, when the scene opens, to set the retrieved values to the scene (through blender string properties), and make them usable later through other scripts. The goal is to mirror the pipe values from OpenPype directly in the scene, at each scene opening.

Reply:
Okay. Last question, why manually adding individual script file:
```python
self.launch_context.data.setdefault("python_scripts", []).append(
            os.path.join(custom_script_folder, self.script_file_name)
        )
```
And not all the python file of the hooks folder at once?
Because currently you have to always add a global python variable in your hook files named `script_file_name` only used to add the script.

This is maybe out of scope for this PR, just asking.

Reply:
We have a 1:1 relation between the hooks and the startup scripts. At first I implemented a solution with only one hooks that was configured to automatically add all custom scripts to the `launch_context.data`. But I thought it wasn't a great solution, as it didn't allows us to personnalize the startup scripts we want to trigger and to separate the responsabilities in a correct way (Each hook has a specific role, which allow us to have a better understanding of all the "hidden" mechanism that could be launched at startup, as could sometimes be a bit opaque)

Reply:
Hum.. ok.
But in that case you should add if not a comment, at least a check if the linked script exists, somthing like (in all PreHook file):
```python
script_folder = Path(custom_script_folder)
script_file = script_folder.joinpath(self.script_file_name)
if not script_file.exists() or not script_file.is_file():
    raise ...
```

And btw you should start using pathlib (a part of the Python standard library since Python 3.4) with the **Path** class (fully support with everywhere in Python since Python 3.8) it's way more robust to handle paths than any other lib or package.

Reply:
The check has been added, and I updated the two hooks to use pathlib instead of os.